### DISCUSSION TITLE
词汇讨论

### DISCUSSION THREAD

**Discussion by QC-L** [reactjs/zh-hans.react.dev#611] [Ideas]
关于翻译过程中，出现多次的词语或句子请在这里进行评论，一旦确认，会补充到词汇表中

**Comment by ChelesteWang** [comment]
In this chapter
应该是不是可以统一一下（目前看到有本章节，本章，本节三种）

**Reply by qinhua** [reply]
之前不是说 `what's next` 译做 `下节预告`，这样的话不就应该都统一为 `节` 吗？
不过现在确实挺犹豫的。

**Reply by QC-L** [reply]
规范下的话，用 **本章节** 会好些

**Comment by dahui4dev** [comment]
Linting 是不是保留，不用翻译比较好？

**Reply by dahui4dev** [reply]
比如  `Code linters` 是不是不翻译或者 翻译之后保留原文：代码过滤器（Code linters）这样？

**Reply by dahui4dev** [reply]
或者参照 ESLint 里的直接翻译为：代码检查工具。

**Reply by watonyweng** [reply]
建议翻译为`代码检查工具`

**Reply by QC-L** [reply]
Linting 要看语境了，可以给出具体例子

**Reply by dahui4dev** [reply]
比如这里我 `Code linters` 做了保留。
![image](https://user-images.githubusercontent.com/6160798/139873131-c6580aa2-3c29-4d2b-9361-0b3874fa175d.png)

**Comment by daochouwangu** [comment]
modern 如何翻译，翻译为`现代/现代化` 感觉有点怪，翻译成 `最新的` 又偏口语
例句：
```
If you use modern JavaScript syntax,
```
```
In modern React without class components, the equivalent code would call useRef instead:
```

**Reply by awxiaoxian2020** [reply]
参考 https://zh.javascript.info/
我觉得都可，`最新的` 我觉得没有口语化的问题

**Comment by zifeiyu666** [comment]
哈喽，大家，memoization 这个词是要怎么翻译啊？感觉不管怎么翻译都没有英文这个单词表达的更加清晰？

**Reply by fwqaaq** [reply]
有整段内容吗

**Reply by zifeiyu666** [reply]
![WX20230410-142347](https://user-images.githubusercontent.com/23133123/230840636-b32a33f9-ce48-4125-8bbe-738c8c7d82a6.png)

**Reply by zifeiyu666** [reply]
![WX20230410-142318](https://user-images.githubusercontent.com/23133123/230840691-ec35f906-ec60-4e03-ba7a-f27825971161.png)

**Reply by zifeiyu666** [reply]
类似于上面这种

**Reply by fwqaaq** [reply]
wiki 百科的翻译，较久未更新：<https://zh.wikipedia.org/wiki/%E8%AE%B0%E5%BF%86%E5%8C%96>

**Reply by zifeiyu666** [reply]
但是这里如果翻译成『记忆化』或者『记忆』之类的很难让读者第一时间想到指的就是 memoization 这种技术

**Reply by fwqaaq** [reply]
如果不知道 memoization 名词的人同样也并不知道这个技术，然而记忆化能有恰当的联想，这个看译者个人选择

**Reply by fwqaaq** [reply]
如果你想要翻译同时保留英文，你可以在开头这样做 -> 记忆化（memoization）

**Comment by awxiaoxian2020** [comment]
关于 fallback 确实应该翻译为宜。但“回落”“”降级“或“备用”似乎并不是一个好的译法，因为原文中既有动词也有名词的用法，我们需要考虑一种更佳的译法。 @fwqaaq

**Reply by fwqaaq** [reply]
[这里很显然是名词](https://github.com/reactjs/zh-hans.react.dev/pull/1131#discussion_r1172181616)，fallback 是个很常见的词，它可以用于 OS 的字体设置  `fallback font`，这里常常是备用、后备；在某些网络相关的文档中，常常用作回落、回退；或者 `fallback position`，退路。

这是英文词典的解释

 >1 an alternative plan that may be used in an emergency: teaching was a last resort, a fallback.
 >
 > 2 a reduction or retreat: the offering will hit the market after a fallback from record highs.

刚开始我认为这是网络中的专有名词，在之后见到很多地方都会用它，它应该还是很常见的

**Reply by awxiaoxian2020** [reply]
我的意思是在中文表达中”回落“不适合作为名词。

**Reply by fwqaaq** [reply]
我并没有说一定要回落，而是不翻译不好

**Reply by awxiaoxian2020** [reply]
> 我并没有说一定要回落，而是不翻译不好

你注意下我在这个 discussion 开始怎么说的hhh

**Comment by fwqaaq** [comment]
本来就这几个意思最贴切，你把都排除了，那还能怎么整？
（除了降级）而且原文明显是省略了component，fallback 一般做定语，一个 fallback 怪不怪？

**Reply by awxiaoxian2020** [reply]
> 本来就这几个意思最贴切，你把都排除了，那还能怎么整？
> （除了降级）而且原文明显是省略了component，fallback 一般做定语，一个 fallback 怪不怪？

我的意思是直接翻译不妥，应将省略的成分补充，那你认为如果省略的是组件，应该用什么作为定语呢？

**Reply by fwqaaq** [reply]
a fallback component

**Reply by awxiaoxian2020** [reply]
> a fallback component

我是说这个你要怎么翻译，
是回落的组件，还是备用的组件，还是什么样的组件

**Reply by fwqaaq** [reply]
在这里：https://github.com/reactjs/zh-hans.react.dev/pull/1131#discussion_r1168125832

**Reply by fwqaaq** [reply]
当然回落不行，还有一些其他词....，并没有一定要哪个词，只是这样更好

**Reply by awxiaoxian2020** [reply]
> 当然回落不行，还有一些其他词....，并没有一定要哪个词，只是这样更好

你没有直接回答我的问题啊……不行的话，那你准备用哪个词？我没有合适的词，所以暂时不翻译……

**Comment by fwqaaq** [comment]
备用/回退……

**Reply by awxiaoxian2020** [reply]
目前中文互联网上我暂未搜索到 fallback component 的中文使用，所以我的建议是翻译成备用组件并标注英文。

**Comment by fwqaaq** [comment]
你可以参考 vue3 的翻译

https://vuejs.org/guide/built-ins/suspense.html

**Reply by awxiaoxian2020** [reply]
所以你的意思是翻译为“后备”吗？

**Reply by fwqaaq** [reply]
这里我觉得这几个都可以，看贡献者意愿

**Comment by jrr997** [comment]
See more examples below 在41个文件中出现了，应该补充到词汇表中，建议翻译为 请看下方更多示例。

**Comment by kanbykanby** [comment]
escape hatches 为什么翻译成“应急方案”呢？感觉这个翻译很呆板和不人性化。我注意到繁体中文版本的翻译是：”逃脫出口“。感觉比应急方案更雅一些。应急方案有点像AI翻译，会有点无聊，然后它也没那么“急”...
像 Effect 就是一种 “逃脱出口”，它让你可以 “跳出/逃出” react 和外部状态同步，比如网络之类的。也不算一种应急的东西。表达了一种 escape 逃脱的感觉。
https://zh-hans.react.dev/learn/escape-hatches

**Reply by kanbykanby** [reply]
@QC-L @Huxpro 你们觉得呢？😆

**Reply by kanbykanby** [reply]
Effects are an escape hatch from the React paradigm.
Effect 是 React 范式中的一个逃脱出口。
Effect 是 React 范式中的一个应急方案。
哪个好一些？

**Reply by BetterZxx** [reply]
翻译成应急方案确实有些问题，是否可翻译成-逃离方案/方法。
useEffect 中也存在这个词：[https://react.dev/reference/react/useEffect#wrapping-effects-in-custom-hooks](https://react.dev/reference/react/useEffect#wrapping-effects-in-custom-hooks)

**Reply by Huxpro** [reply]
@shezhangzhang 这种本来就是个英文比喻的词确实比较难翻译，我同意「应急」不是特别适合，而「逃脱出口」又可能过于接近一个比喻了，类似 @BetterZxx，我觉得「逃脱方案」可能还不错，甚至是「“逃脱”方案」（永远都打上引号表示一种比喻）

除了现在对内使用的词汇表之外，如果可以在文档里加一个对外的术语解释可能会对用户更加友好一些。可以详细去解释：
> 在技术圈子中，"Escape Hatch" 一般用于指代，某些框架或系统在使用的过程中，为了能够支持高度个性化的需求场景，提供的一些"最后的手段"或者"逃生口"。这些逃生口可以让开发者绕过系统中某些默认行为，以便于他们能够更好的定制和扩展系统。
在 React 和其他一些框架的官方文档中，"Escape hatch" 一般用于指代一些高度个性化的参数或者 API，这些 API 的使用允许开发者更改组件的默认属性，以及更改React或其他框架的一些内部行为，或者在特殊情况下规避某些限制或者 bug。例如，React 官方文档中的 "Escape Hatch Props" 就是指提供在组件中使用的一些逃生口属性，这些属性在通常情况下不应该被频繁使用，但在某些特殊场景下被发现是非常有用的。
> -- chatGPT

**Reply by kanbykanby** [reply]
同意 `escape hatches` => `逃脱方案` +1 👍

**Reply by kanbykanby** [reply]
导航里面没有加引号，因为导航栏中这个词，在那里显示引号可能破坏导航栏的整体性，文中加不加引号建议视情况而定。

**Reply by awxiaoxian2020** [reply]
我在这里想的是，是否需要翻译成“逃脱”，“逃”感觉是一种面对糟糕情形的动作

**Reply by kanbykanby** [reply]
因为英文用的是escape，就是逃脱逃跑的意思，可能是形容这些行为（如effect）逃出了react的默认控制，是一种给你开的逃脱方案！感觉这里的逃脱有点儿俏皮的意思👀

**Reply by earthaYan** [reply]
翻译为  绕过方案 会不会更适合

**Reply by zhenmang** [reply]
翻译为 脱离束缚 也比较贴切。提到的文中两处可做如下翻译：
1. 原文中 Effects are an [“escape hatch”:](https://react.dev/learn/escape-hatches) 可译为 Effect是一种脱离束缚的手段。
2. 原文中 you’ll learn the escape hatches 可译为 你将学习脱离束缚的方法。
整体上都是为了表达脱离react范式的束缚，灵活的处理【某些场景下因响应式带来的问题】。

**Comment by earthaYan** [comment]
sandbox示例代码中的 TODO  是否应该翻译？
1. 如果翻译：译成 待办项 是否合适
2. 如果不翻译：希望能加入列表

**Reply by earthaYan** [reply]
以及代码中出现的Avoid和 Great ,个人倾向于同Good,Bad一样不翻译

**Reply by loveloki** [reply]
TODO 我感觉真没必要翻译吧，平常写代码也经常写个注释加 TODO。。。没有任何理解问题

**Comment by earthaYan** [comment]
*regardless* of the specific interactions performed by the user.Whether the user has only opened your app, selected a different room, or navigated to another **_screen_** and back/, your Effect ensures that the component will *remain synchronized* with the currently selected room, and will re-connect whenever it's necessary。
 
大佬们，请教下这句中的screen 翻译成什么比较合适呢？字典释义是 屏幕，但是这样翻译感觉有点奇怪，我理解这里想表达的意思是从其他页面返回到聊天室页面

**Comment by PrinOrange** [comment]
DevTools也加进去吧

**Comment by PrinOrange** [comment]
另外还有“In development”这种词，我在文档中见过“开发环境”、“开发模式”下两种翻译

**Comment by Huxpro** [comment]
@QC-L 我们可以参考 Vue 社区的 https://github.com/vuejs-translations/docs-zh-cn/wiki/%E7%BF%BB%E8%AF%91%E9%A1%BB%E7%9F%A5 把 #2 贴到 wiki，我觉得每一个的讨论可以单独开 issue？这样比较好确定一个术语的翻译是否有「结论」了

**Reply by QC-L** [reply]
可以，那就新建一个 wiki 来承接。

**Reply by Yucohny** [reply]
> 可以，那就新建一个 wiki 来承接。

我来吧。

**Comment by Yucohny** [comment]
See https://github.com/reactjs/zh-hans.react.dev/wiki/The-Glossary-Sheet.

**Reply by Huxpro** [reply]
@Yucohny thanks! Let's make sure previous discussions are either captured in the Wiki if it's resolved (potentially with the issue that resolved it) or creating new issues for the open ones (say, the ones in this thread)

**Reply by Huxpro** [reply]
@Yucohny notably, escape hatches, fallback, 等等，包括所有 intentionally 不翻译的也都可以列一下，可以让大家 revisit
